walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -98,68 +93,85 @@ public boolean optionalAttendeesOnly(Event event, Collection<String> optionalAtt
+        Set<String> eventAttendees = event.getAttendees();
+        for (String optAttendee : optionalAttendees) {
+            if (eventAttendees.contains(optAttendee)) {
+                return true;
             }
-            count++;
-        }
-        int firstStart = eventOneTimeRange.start();
-        int firstEnd = eventOneTimeRange.end();
-        int secondStart = eventTwoTimeRange.start();
-        int secondEnd = eventTwoTimeRange.end();
-
-        if (eventOneTimeRange.contains(eventTwoTimeRange)) {
-            Collection<TimeRange> containedTimes = Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, firstStart, false),
-            TimeRange.fromStartEnd(firstEnd, TimeRange.END_OF_DAY, true));
-            return containedTimes;
         }
-        else if (eventTwoTimeRange.contains(eventOneTimeRange)) {
-            Collection<TimeRange> containedTimes = Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, secondStart, false),
-            TimeRange.fromStartEnd(secondEnd, TimeRange.END_OF_DAY, true));
-            return containedTimes;
-        }
-        else if (eventOneTimeRange.overlaps(eventTwoTimeRange)) {
-            Collection<TimeRange> overlapTimes =
-            Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, firstStart, false),
-            TimeRange.fromStartEnd(secondEnd, TimeRange.END_OF_DAY, true));
-            return overlapTimes;
+        return false;
+    }
+
+    private Collection<TimeRange> getAllTimes (List<Event> events, MeetingRequest request) {
+        Collection<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+        long requestedDuration = request.getDuration();
+        boolean last = false;
+
+        Event firstEvent = events.get(0);
+        TimeRange firstStart = firstEvent.getWhen();
+        int start = firstStart.start();
+        TimeRange before = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, start, false);
+        if (TimeRange.START_OF_DAY != start) {
+            possibleTimes.add(before);
         }
-        else if (firstStart == 0 && secondEnd == 1440) {
-            int actualDuration = secondStart - firstEnd;
-            if (duration > actualDuration) {
-                //not enough time
-                Collection<TimeRange> noTime = Arrays.asList();
-                return noTime;
+
+        for (int i = 1; i < events.size(); i++) {
+            TimeRange curEvent = events.get(i-1).getWhen();
+            TimeRange nextEvent = events.get(i).getWhen();
+            int curEventEnd = curEvent.end();
+            int nextEventStart = nextEvent.start();
+            int nextEventEnd = nextEvent.end();
+
+            if (!curEvent.overlaps(nextEvent)) {
+                int timeBetweenEventsInt = nextEventStart - curEventEnd;
+                if (timeBetweenEventsInt >= requestedDuration) {
+                    possibleTimes.add(TimeRange.fromStartEnd(curEventEnd, nextEventStart, false));
+                } 
+            }
```

Comment:
In general, I think it looks better to break lines such that:

```
if () {
} else if () {
} else {
}
```

Reply:
agreed- fixed